artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AmqpCoreConverter.java

```diff
@@ -444,7 +446,7 @@ private static void encodeUnsupportedMessagePropertyType(ServerJMSMessage jms, S
       }
    }
 
-   private static void setProperty(javax.jms.Message msg, String key, Object value) throws JMSException {
+   private static void setProperty(ServerJMSMessage msg, String key, Object value) throws Exception {
```

Comment:
-1 throwing raw exception

Reply:
This is only used on conversions.. there's no need for anything specifically here...  This is just throwing whatever happens underneath.

Reply:
Then maybe have a single base exception for these.  At least that way can still differentiate between those and other possible unexpected exceptions that could arise from a bug.